Implement collision impulse in Godot Physics 3D

CherrySodaPop opened the pull request:
Going off of #39114 and the outdated 3D physics state docs:
``Impulse created by the contact. Only implemented for Bullet physics.``

I've implemented the returning of the impulse forces by contacts into GodotPhysics3D
I had originally feared this would break compatibility but considering the function didn't work at all and simply returned ``0.0`` it shouldn't cause any issues with it now returning a Vector3 of the impulse forces.

rburing reviewed (approved):
Looks good. Is `c.acc_impulse_A == -c.acc_impulse_B`? Then it doesn't need two variables (if reduced to a single variable then a comment could explain it, similarly to the other members).

YuriSizov commented:
Make sure to amend the commit message to state what this PR does exactly. I've changed the title for this pull request for a reference, but feel free to update it as well, if I've misunderstood what this PR does.

See [this document](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-messages-with-readability-in-mind) for guidelines.

CherrySodaPop committed:
Implement collision impulse in Godot Physics 3D

CherrySodaPop commented:
> Looks good. Is `c.acc_impulse_A == -c.acc_impulse_B`? Then it doesn't need two variables (if reduced to a single variable then a comment could explain it, similarly to the other members).

Condensed and added a comment!
Also made me notice that during the pre-solve the impulse wasn't being properly set, fixed also.

> Make sure to amend the commit message to state what this PR does exactly. I've changed the title for this pull request for a reference, but feel free to update it as well, if I've misunderstood what this PR does.

Corrected the commit message.

akien-mga commented:
Thanks!
